docs/kb/semgrep-supply-chain/sss-python-lockfiles.md

```diff
@@ -0,0 +1,294 @@
+---
+description: Generation of Python lock files to run SSC scans successfully.
+tags:
+  - Semgrep Supply Chain
+  - Python
+  - Knowledge Base
+  - Lock files
+  - requirements.txt
+  - Pipfile.lock
+  - Poetry.lock
+---
+
+# Introduction
+
+Python has different mechanisms to get all packages used in a project. The most known file is requirements.txt, a "simple" txt file where you define all the packages and the versions your project needs. 
```

Comment:
```suggestion
Python has different mechanisms to detect all packages used in a project. 

Semgrep Supply Chain can use any of the following methods to correctly parse your dependencies and detect vulnerabilities:

* `requirements.txt`
* `Pipfile.lock`
* Poetry
```

We mention that requirements.txt is the most common file used (in line 21) so we can remove it here. I just shortened the text and created a list for easier skimming.